### PR TITLE
Change BC template to a form

### PR DESCRIPTION
**Pull request by gewarren**
See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms.

### CODE REVIEW

**Comment by gewarren** [comment]
@Youssef1313 The docs verifier should ignore deletions in the .github folder.

No redirection is found for '.github/ISSUE_TEMPLATE/dotnet-breaking-change.md'.

File: .github/ISSUE_TEMPLATE/breaking-change.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+name: ".NET breaking change"
+desription: Report a change in .NET that breaks something that worked in a previous version. Intended mostly for product-team use.
```
Comment (BillWagner) [inline .github/ISSUE_TEMPLATE/breaking-change.yml:2]: Hmmm.  When I try to view this on GitHub using "view file", it fails, saying that "description isn't valid". Any ideas?

File: .github/ISSUE_TEMPLATE/breaking-change.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    validations:
+      required: true
+  - type: dropdown
+    id: version
+    attributes:
+      label: Version
+      description: What version of .NET introduced the breaking change?
+      options:
+        - .NET 6 Preview 6
+        - .NET 6 Preview 7 (Default)
```
Comment (BillWagner) [inline .github/ISSUE_TEMPLATE/breaking-change.yml:22]: Will this require regular edits? If so, was it intentional?

**Review by BillWagner — APPROVED** [review]
This is a great idea @gewarren 

I had a couple questions, and then it's ready to :shipit:

File: .github/ISSUE_TEMPLATE/breaking-change.yml (hunk 1 above)
Comment (Youssef1313) [reply .github/ISSUE_TEMPLATE/breaking-change.yml:2]: This should fix https://github.com/dotnet/docs/pull/25161/files#r670465042.


```suggestion
description: Report a change in .NET that breaks something that worked in a previous version. Intended mostly for product-team use.
```

File: .github/ISSUE_TEMPLATE/breaking-change.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      description: Describe the recommended action an affected user should take, such as workarounds or examples of code changes.
+    validations:
+      required: true
+  - type: dropdown
+    id: area
+    attributes:
+      label: Feature area
+      description: What feature area(s) does the change affect?
+      options:
+        - .NET 6 Preview 6- [ ] ASP.NET Core
```
Comment (Youssef1313) [inline .github/ISSUE_TEMPLATE/breaking-change.yml:74]: This looks like a typo?

```suggestion
        - .NET 6 Preview 6
        - ASP.NET Core
```

File: .github/ISSUE_TEMPLATE/breaking-change.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    attributes:
+      label: Recommended action
+      description: Describe the recommended action an affected user should take, such as workarounds or examples of code changes.
+    validations:
+      required: true
+  - type: dropdown
+    id: area
+    attributes:
+      label: Feature area
+      description: What feature area(s) does the change affect?
```
Comment (Youssef1313) [inline .github/ISSUE_TEMPLATE/breaking-change.yml:72]: The dropdown doesn't support multiple choices.

File: .github/ISSUE_TEMPLATE/breaking-change.yml (hunk 2 above)
Comment (gewarren) [reply .github/ISSUE_TEMPLATE/breaking-change.yml:22]: Yes, and yes. Or they can select "Other" in a pinch.

File: .github/ISSUE_TEMPLATE/breaking-change.yml (hunk 3 above)
Comment (gewarren) [inline .github/ISSUE_TEMPLATE/breaking-change.yml:74]:
```suggestion
        - [ ] ASP.NET Core
```

**Pull request merged**